python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1300,6 +1301,32 @@ def _test_write_to_dataset_with_partitions(base_path,
     tm.assert_frame_equal(output_df, input_df)
 
 
+def test_write_metadata_with_without_filesystem(tempdir):
```

Comment:
A `write_metadata` specific test is in `parquet/test_metadata.py`, so maybe move this there?